pkg/theia/commands/policy_recommendation_run.go

```diff
@@ -0,0 +1,476 @@
+		if err != nil {
+			return err
+		}
+		recoJobArgs = append(recoJobArgs, "--rm_labels", strconv.FormatBool(excludeLabels))
+
+		toServices, err := cmd.Flags().GetBool("to-services")
+		if err != nil {
+			return err
+		}
+		recoJobArgs = append(recoJobArgs, "--to_services", strconv.FormatBool(toServices))
+
+		executorInstances, err := cmd.Flags().GetInt32("executor-instances")
+		if err != nil {
+			return err
+		}
+		if executorInstances < 0 {
+			return fmt.Errorf("executor-instances should be an integer >= 0")
+		}
+		sparkResourceArgs.executorInstances = executorInstances
+
+		driverCoreRequest, err := cmd.Flags().GetString("driver-core-request")
+		if err != nil {
+			return err
+		}
+		matchResult, err := regexp.MatchString(k8sQuantitiesReg, driverCoreRequest)
+		if err != nil || !matchResult {
+			return fmt.Errorf("driver-core-request should conform to the Kubernetes resource quantity convention")
+		}
+		sparkResourceArgs.driverCoreRequest = driverCoreRequest
+
+		driverMemory, err := cmd.Flags().GetString("driver-memory")
+		if err != nil {
+			return err
+		}
+		matchResult, err = regexp.MatchString(k8sQuantitiesReg, driverMemory)
+		if err != nil || !matchResult {
+			return fmt.Errorf("driver-memory should conform to the Kubernetes resource quantity convention")
+		}
+		sparkResourceArgs.driverMemory = driverMemory
+
+		executorCoreRequest, err := cmd.Flags().GetString("executor-core-request")
+		if err != nil {
+			return err
+		}
+		matchResult, err = regexp.MatchString(k8sQuantitiesReg, executorCoreRequest)
+		if err != nil || !matchResult {
+			return fmt.Errorf("executor-core-request should conform to the Kubernetes resource quantity convention")
+		}
+		sparkResourceArgs.executorCoreRequest = executorCoreRequest
+
+		executorMemory, err := cmd.Flags().GetString("executor-memory")
+		if err != nil {
+			return err
+		}
+		matchResult, err = regexp.MatchString(k8sQuantitiesReg, executorMemory)
+		if err != nil || !matchResult {
+			return fmt.Errorf("executor-memory should conform to the Kubernetes resource quantity convention")
+		}
+		sparkResourceArgs.executorMemory = executorMemory
+
+		kubeconfig, err := ResolveKubeConfig(cmd)
+		if err != nil {
+			return err
+		}
+		clientset, err := CreateK8sClient(kubeconfig)
+		if err != nil {
+			return fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+		}
+
+		waitFlag, err := cmd.Flags().GetBool("wait")
+		if err != nil {
+			return err
+		}
+
+		err = PolicyRecoPreCheck(clientset)
+		if err != nil {
+			return err
+		}
+
+		recommendationID := uuid.New().String()
+		recoJobArgs = append(recoJobArgs, "--id", recommendationID)
+		recommendationApplication := &sparkv1.SparkApplication{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "sparkoperator.k8s.io/v1beta2",
+				Kind:       "SparkApplication",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "policy-recommendation-" + recommendationID,
```

Comment:
We may need a shorter name here as the UI service seems to fail to start if the name is too long. Find the following errors in the spark operator log:
```
I0604 01:07:15.780425      10 controller.go:263] Starting processing key: "flow-visibility/policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8"
I0604 01:07:15.780611      10 sparkui.go:282] Creating a service policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8-ui-svc for the Spark UI for application policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8
I0604 01:07:15.780861      10 event.go:282] Event(v1.ObjectReference{Kind:"SparkApplication", Namespace:"flow-visibility", Name:"policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8", UID:"9a99144f-3ef3-4ee2-bc81-57232f48cb2d", APIVersion:"sparkoperator.k8s.io/v1beta2", ResourceVersion:"18587", FieldPath:""}): type: 'Normal' reason: 'SparkApplicationAdded' SparkApplication policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8 was added, enqueuing it for submission
E0604 01:07:15.821772      10 controller.go:670] failed to create UI service for SparkApplication flow-visibility/policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8: Service "policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8-ui-svc" is invalid: metadata.name: Invalid value: "policy-recommendation-07fdd999-a7ab-4569-b918-8330237e32d8-ui-svc": must be no more than 63 characters
```

Reply:
Thanks Yanjun pointing out this. I propose we could change the job name to `pr-<id>`. I plan to make this change tomorrow if no objections.

Reply:
ops, thanks for spotting it @yanjunz97  - let's wait for a PR update from @dreamtalen

Reply:
Updated.